packages/cbioportal-utils/src/hotspot/CancerHotspotsUtils.spec.ts

```diff
@@ -209,6 +376,77 @@ describe('CancerHotspotsUtils', () => {
         });
     });
 
+    describe('isRecurrentHotspot', () => {
+        it('checks if a mutation is a hotspot mutation by isRecurrentHotspot()', () => {
+            assert.isTrue(
+                isRecurrentHotspot(hotspotMutation1, hotspotIndex),
+                'TP53 R273C should be a recurrent hotspot mutation.'
+            );
+
+            assert.isTrue(
+                isRecurrentHotspot(hotspotMutation2, hotspotIndex),
+                'TP53 R273A should be a recurrent hotspot mutation.'
+            );
+
+            assert.isTrue(
+                isRecurrentHotspot(hotspotMutation3, hotspotIndex),
+                'PIK3CA R38H should be a recurrent hotspot mutation.'
+            );
+
+            assert.isFalse(
+                isRecurrentHotspot(notHotspotMutation, hotspotIndex),
+                'SMURF1 R101F should not be a recurrent hotspot mutation.'
+            );
+
+            assert.isFalse(
+                isRecurrentHotspot(hotspot3dMutation, hotspotIndex),
+                'SMURF1 R101N should not be a recurrent hotspot mutation.'
+            );
+        });
+    });
+
+    describe('is3dHotspot', () => {
+        it('checks if a mutation is a hotspot mutation by is3dHotspot()', () => {
+            assert.isFalse(
+                is3dHotspot(notHotspotMutation, hotspot3dIndex),
+                'SMURF1 R101F should not be a 3d hotspot mutation.'
+            );
+
+            assert.isTrue(
+                is3dHotspot(hotspot3dMutation, hotspot3dIndex),
+                'SMURF1 R101N should be a 3d hotspot mutation.'
+            );
+        });
+    });
+
+    describe('isSpliceHotspot', () => {
+        it('checks if a mutation is a hotspot mutation by isSpliceHotspot()', () => {
+            assert.isFalse(
+                isSpliceHotspot(notHotspotMutation, spliceHotspotIndex),
+                'MET X1028_splice should not be a splice hotspot mutation.'
+            );
+
+            assert.isTrue(
+                isSpliceHotspot(hotspotSpliceMutation, spliceHotspotIndex),
+                'SMURF1 R101N should be a splice hotspot mutation.'
+            );
+        });
+    });
+
+    describe('isNon3dHotspot', () => {
+        it('checks if a mutation is a hotspot mutation by isNon3dHotspot()', () => {
+            assert.isTrue(
+                isNon3dHotspot(hotspotSpliceMutation, spliceHotspotIndex),
+                'SMURF1 R101N should be a non-3d(splice) hotspot mutation.'
```

Comment:
@leexgh this one I don't fully understand. `SMURF1 R101N` means amino acid R changes to N and by definition is not splice but missense. It also does not seem to be hotpot in https://www.cancerhotspots.org/#/home or https://www.3dhotspots.org/#/home. So shouldn't this return false?

Reply:
@inodb oops I messed up the description message😢 , the logic and tests are correct - it tests `MET X1010`, and this message should be 
```
'MET X1010 should be a non-3d(splice) hotspot mutation.'
```

Reply:
@inodb I think this is a made up mock data that I added long time ago, that's why the gene and amino acid combination do not make any sense :)

Reply:
ah ok that makes sense, i should've checked the code as well 😅